### PR TITLE
add self-hosted wallet generator

### LINKED ISSUE #2121: remove/replace moneroaddress.org
* Core have been asked to host the site. Ercicione suggested 'generator.getmonero' which is perfect

user ofrnxmr noticed we're linking to moneroaddress dot org. This site 'changed hands' some time ago and should be removed/replaced entirely. 

the easiest solution would be a string replace it with a site owned/controlled by a member of core https://xmr.llcoins.net/ (as this allows users to generate a random seed)

or better: we get a member of core to host the original version of this html file: https://github.com/moneromooo-monero/monero-wallet-generator and directly link to that instead (using a domain we own).

### PR DESCRIPTION
**Pull request by erciccione**
This is @moneromooo-monero's wallet generator (https://github.com/moneromooo-monero/monero-wallet-generator) adapted for getmonero. It will be available at `getmonero.org/generator`.

Changes:

- changed structure to make it more mobile friendly
- removed some unused css 
- added getmonero's headers for SEO  and opengraph views

I touched the file as less possible to make reviewing easier. It could be cleaned up further. I stopped when realized there is too much stuff entangled with each other (printable page, mobile, etc) and would take too long. Probably not worth it.

The file assumes there is a signed version of the page hosted in the same folder and signed with @binaryFate's key.

closes #2121

### CODE REVIEW

**Comment by erciccione** [comment]
Added fixes suggested by ofrnxmr and added dummy signed page and dummy zip file

**Review by nahuhh — CHANGES_REQUESTED** [review]

File: _i18n/en/resources/user-guides/securely_purchase.md
Comment (erciccione) [reply _i18n/en/resources/user-guides/securely_purchase.md]: Added.

**Comment by erciccione** [comment]
CI fixed by #2135

**Review by nahuhh — CHANGES_REQUESTED** [review]

File: resources/wallet-generator/wallet-generator.html
Comment (plowsof) [inline resources/wallet-generator/wallet-generator.html:8450]:
```suggestion
            Made by moneromooo. Original CSS by antanst (Updated and adapted for Getmonero.org by ErCiccione), crypto fixes by luigi111. Based on code from <a href="https://mymonero.com/" target="_blank">MyMonero</a>. Copyright notices in the source.
```
no one else has a 2nd handle listed and makes it more clear what has been done.

just checking that removing the donation address of mooo has been agreed upon with them? i mean i understand there are reasons for that e.g. security

**Comment by plowsof** [comment]
tried to comment a suggestion for credits, didn't go well so ill just add it here: no one has a 2nd handle, and improves clarity as to what updates where done

orig:
``` html
Made by moneromooo. Original CSS by antanst, crypto fixes by luigi111. Based on code from <a href="https://mymonero.com/" target="_blank">MyMonero</a>. Updated and adapted for Getmonero.org by ErC (ErCiccione). Copyright notices in the source.
```
suggestion:
``` html
Made by moneromooo. Original CSS by antanst (Updated and adapted for Getmonero.org by ErCiccone), crypto fixes by luigi111. Based on code from <a href="https://mymonero.com/" target="_blank">MyMonero</a>. Copyright notices in the source.
```

and i just want to clear that removing mooos donation address was agreed on? i understand there are reasons for this e.g. security

File: resources/wallet-generator/wallet-generator.html
Comment (hinto-janai) [inline resources/wallet-generator/wallet-generator.html:1]: I think this should be rebased so that the [original](https://github.com/moneromooo-monero/monero-wallet-generator/blob/master/monero-wallet-generator.html) is added and then the changes.

It would be a lot easier to see the diffs.

Comment (erciccione) [reply resources/wallet-generator/wallet-generator.html:1]: This wouldn't help if the goal is to make reviewing easier or make sure the file being added here is not malicious. The diff between the two files can be seen locally by simply running `diff file 1 file 2`. Any other approach would be unnecessary and will not guarantee that after the rebase (because there will be an unnecessary commit that will have to be removed after the review) the file it's still the same that was reviewed.

**Comment by erciccione** [comment]
> and i just want to clear that removing mooos donation address was agreed on? i understand there are reasons for this e.g. security

@plowsof  I removed it and pointed people to his version, which includes the donation address. I didn't think it made sense to add the address on getmonero's version and we would also always need to make sure the address is still used.

File: resources/wallet-generator/wallet-generator.html
Comment (hinto-janai) [reply resources/wallet-generator/wallet-generator.html:1]: Do commits always have to be squashed? I was thinking something like:
```
This_PR:
    commit_1: add wallet-generator.html
    commit_2: add changes
    commit_3: [...]
```
If it were like this, the diffs would be viewable in this repo for anyone to see, for all time.

Comment (erciccione) [reply resources/wallet-generator/wallet-generator.html:1]: > Do commits always have to be squashed? 

If they are pointless and make git's history heavy (especially since it's a large file), yes. Let's keep the standard git workflow, more so since the only benefit would be to see a diff on github instead of locally.

**Comment by erciccione** [comment]
Force pushed to use `ErCiccione` instead of `ErC (ErCiccione)` in the credits, as suggested by selsta.

File: resources/wallet-generator/monero-wallet-generator.html
Comment (hinto-janai) [inline resources/wallet-generator/monero-wallet-generator.html:8400]: This used to be blue, not pink. Was the choice to make it the same color as the seed intentional? (Public info = green, private info = pink). If so, seems good.

File: resources/wallet-generator/monero-wallet-generator.html
Comment (hinto-janai) [inline resources/wallet-generator/monero-wallet-generator.html:8435]:
```suggestion
                This step is optional but highly recommended, as it will allow you to verify that this wallet generator hasn't been tampered in any way.
```

File: resources/wallet-generator/monero-wallet-generator.html
Comment (erciccione) [reply resources/wallet-generator/monero-wallet-generator.html:8402]: Yes, the original used three different colours. I thought of simplifying by making visually clear what is safe to share publicly (green) and what is meant to keep secret (pink). Thanks for the review, i'll add your other suggestions tomorrow.

File: resources/wallet-generator/monero-wallet-generator.html
Comment (hinto-janai) [reply resources/wallet-generator/monero-wallet-generator.html]: Nvm, these aren't being renamed (I think), the diff view made it confusing. Although, the deploy preview QR buttons are broken, not sure if that's a deploy thing or not.

**Comment by erciccione** [comment]
Added suggestions from @hinto-janai. Let's wait some days for more reviewers. After we need to get the file signed by @binaryFate or somebody else from core.

File: resources/wallet-generator/monero-wallet-generator.html
Comment (plowsof) [inline resources/wallet-generator/monero-wallet-generator.html:2]: appears when you print. see #2156. an index.html file next to the wallet solves the issue (no need to edit _config.yml anymore)

File: resources/wallet-generator/monero-wallet-generator.html
Comment (plowsof) [inline resources/wallet-generator/monero-wallet-generator.html:8201]:
```suggestion
        <a href="." class="brand-logo center">
```
a dot here will take users back to the base url if viewing it online (e.g. getmonero OR an onion url) - offline users will be taken to the parent directory and view a file explorer where they can click again on the html file) OR we could just remove the clickable/href on the logo all together

Comment (plowsof) [inline resources/wallet-generator/monero-wallet-generator.html:8201]: so the logo is clickable, but does nothing. href="." will return users to the base url (getmonero or onion). offline users will be shown the parent directory / a file explorer. i find it ux+ because i expect to be taken home when i click that logo with the online version

File: resources/wallet-generator/monero-wallet-generator.html
Comment (erciccione) [reply resources/wallet-generator/monero-wallet-generator.html:2]: I'm not sure of what you mean. Would you add an empty index.html file?

Comment (erciccione) [reply resources/wallet-generator/monero-wallet-generator.html]: Not a big fan of adding another file for this, but if it's wanted i'm ok with adding it. Leaving this as unresolved in case there is more feedback. Added in the meantime.

File: resources/wallet-generator/monero-wallet-generator.html
Comment (erciccione) [reply resources/wallet-generator/monero-wallet-generator.html]: Done.

File: resources/wallet-generator/monero-wallet-generator.html
Comment (hinto-janai) [inline resources/wallet-generator/monero-wallet-generator.html:8287]:
```suggestion
            Besides Monero, you can also create offline wallets for <a href="http://www.aeon.cash/" target="_blank">Aeon</a> and <a href="https://www.townforge.net/" target="_blank">Townforge</a>. For what cryptocurrency are you creating a wallet?
```

File: resources/wallet-generator/monero-wallet-generator.html
Comment (hinto-janai) [inline resources/wallet-generator/monero-wallet-generator.html:8439]:
```suggestion
                <li>Download the signature file: <a href="https://raw.githubusercontent.com/monero-project/monero-site/master/resources/wallet-generator/wallet-generator.html.asc" target="_blank">wallet-generator.html.asc</a></li>
```
I think showing the file name that will be downloaded (or something related) is more standard.

**Review by hinto-janai — APPROVED** [review]
Confirming:
- Clicking header redirects correctly
- QR button issue was indeed a Tor browser thing, stock Firefox works fine

The QR codes for Aeon/Townforge don't work, but the original was like that too. I'd actually prefer they were removed completely, as this is a _Monero_ wallet generator adapted for and hosted on _getmonero.org_. I think it would be less confusing for users.

Although, I'm okay with merging as is.

**Comment by erciccione** [comment]
Added @hinto-janai's suggestions. I would say it's ready to be signed by @binaryFate.

**Review by hinto-janai — APPROVED** [review]

**Comment by rbrunner7** [comment]
Does this wait for a second review? Or is it controversial in any way? Or simply fell through the cracks?

I recently wanted to send somebody asking for a paper wallet generator on Reddit to our page on getmonero.org about that and was quite puzzled to still see https://moneroaddress.org/ there ...

File: resources/wallet-generator/monero-wallet-generator.html
Comment (plowsof) [inline resources/wallet-generator/monero-wallet-generator.html:8275]:
```suggestion
                <li><b>Offline:</b> <a href="https://www.getmonero.org/resources/wallet-generator/monero-wallet-generator.zip" target="_blank">Download the zip file</a> and extract it locally. Open the "monero-wallet-generator.html" page and follow the instructions. You can also use the <a href="https://www.getmonero.org/resources/user-guides/securely_purchase.html" target="_blank">dedicated guide on Getmonero</a></li>
```

**Comment by plowsof** [comment]
@erciccione suggested changes is broken on this large file so leaving a comment:

wallet-generator.zip -> monero-wallet-generator.zip
wallet-generator.html -> monero-wallet-generator.html
```html
<li><b>Online:</b> browse this page at getmonero.org/generator and generate a new wallet</li>
<li><b>Offline:</b> <a href="https://www.getmonero.org/resources/wallet-generator/monero-wallet-generator.zip" target="_blank">Download the zip file</a> and extract it locally. Open the "monero-wallet-generator.html" page and follow the instructions. You can also use the <a href="https://www.getmonero.org/resources/user-guides/securely_purchase.html" target="_blank">dedicated guide on Getmonero</a></li>
```

How to review/sign/zip files:

- Step 1: after puling this PR, Perform interactive rebase (rename pick to edit on original wallet generator from moneromooo)
  ```
  git pull monero pull/2131/head
  git rebase -i d9dcbd0~
  edit d9dcbd08 original wallet generator from moneromooo
  ```

- Step 2: Download and import mooos public key
  ```
  cd resources/wallet-generator/
  wget https://raw.githubusercontent.com/monero-project/monero/master/utils/gpg_keys/moneromooo.asc
  gpg --import moneromooo.asc
  ```

- Step 3: Verify the signed file
  ```
  gpg --verify monero-wallet-generator.html.asc 
  ```
- Step 4: Finish the rebase
  ```
  rebase --continue
  ```

- Step 5: Sign the HTML file with your own key (Overwrite existing file)
  ```
  gpg --output monero-wallet-generator.html.asc --armor --detach-sig monero-wallet-generator.html
  ```

- Step 6: Create a ZIP file
  ```
  zip monero-wallet-generator.zip monero-wallet-generator.html monero-wallet-generator.html.asc
  ```
- Step 7: Delete placeholder / mooo's keyfile
  ```
  rm moneromooo.asc 
  rm monero-wallet-generator.placeholder.zip
  ```
Double check the file/folder structure:
  ```
  resources/
  └── wallet-generator/
      ├── index.html
      ├── monero-wallet-generator.html
      ├── monero-wallet-generator.html.asc
      └── monero-wallet-generator.zip
  ```
 - Step 8: Write a cool commit message
  ```
  commit -m "wallet-generator: signed and zipped"
  ```

**Comment by erciccione** [comment]
> Does this wait for a second review? Or is it controversial in any way? Or simply fell through the cracks?

This is only waiting for @binaryFate's signature. it's ready to be merged.

@plowsof About renaming files. This is not only a Monero wallet generator, it also works with other currencies. 

About changing online-offline. it's a very minor edit, but that would cause people to re-review the entire commit, which was already hard to get reviewed, given the complexity and sensitivity of the file. Better not touch this commit now that has been thoroughly reviewed and open successive prs for minor edits.

**Comment by plowsof** [comment]
@erciccione ignore the online/offline html snippet (that was just for renaming files), the generator is fine now, i will update my steps to reflect the renaming of files and approve, thanks.

**Review by plowsof — APPROVED** [review]

**Comment by rbrunner7** [comment]
Still not merged? Paging @binaryFate

It's getting embarrassing that almost a full year after we "lost" moneroaddress . org to a squatter we **still** send people there.

**Pull request merged**
